.github/workflows/builder_docker-based_slsa3.yml

```diff
@@ -0,0 +1,191 @@
+# Copyright The GOSST team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: SLSA Docker-based builder
+
+# TODO: permissions
+
+# TODO: env
+
+defaults:
+  run:
+    shell: bash
+
+###################################################################
+#                                                                 #
+#            Input and output argument definitions                #
+#                                                                 #
+###################################################################
+on:
+  workflow_call:
+    inputs:
+      builder-image:
+        description: >
+          The OCI image name of the builder image responsible for 
+          creating the artifact. This must not include a tag or digest.
+        required: true 
+        type: string
+      builder-digest:
+        description: >
+          The OCI image digest of the builder-image. 
+          This must be specified in order to fully specify the builder 
+          image. The image digest is of the form '<alg>:<digest>'
+        required: true 
+        type: string
+      config:
+        description: >
+          Path to a configuration file containing a command that the builder
+          image should be invoked with and optionally, docker options
+          and environment variables.
+        required: true 
+        type: string
+      output:
```

Comment:
Let's use a common output across all builders. We need to decide the format, and what we include: artifacts, provenance files, etc including where to download them from (thru download-artifact Action).

We can align this format with the BYOB framework as well: once we have it, all our internal builders can also use the framework.

Not needed for this PR.

Reply:
This is the build output produced by the container image execution, i.e. the subject that needs to be measured. (Not the uploaded artifacts). Maybe I should rename this to builder-output?

Reply:
what do you mean by "measure"? Is it the list of artifacts to include in the attestation? If so, shall we remove this option to start with (image run should be able to make a cp of artifacts it wants attestation for in a predefined directory).

Reply:
Yeah exactly -- we can start with pre-declaring where the output should be, e.g. an output folder like `output`.

Reply:
SG.